README.md

```diff
@@ -159,6 +159,19 @@ Identity type, for example, could provide `Id` as its type representative:
 If a type provides a type representative, each member of the type must have
 a `constructor` property which is a reference to the type representative.
 
+
+## `forAll`
+
+[Parametric polymorphism][pp] is essential to these algebras.  When the rules
+for any type are specified as applied to all values (`forall a b`) of the 
+given generic types (here `a` and `b`), the rule must hold for any JavaScript
```

Comment:
I'm wondering how can we make restrictions polymorphism implies more clear here.
What you think on something like this:(bold is for highlighting the addition):

> Parametric polymorphism is essential to these algebras. When the rules
for any type are specified as applied to all values (`forall a b`) of the
given generic types (here `a` and `b`), the rule must hold for any **possible**
JavaScript value ~~unless otherwise specified~~. 
**Since a parametrically polymorphic value does not "know" anything about the
type variables, it must behave the same regardless of its type. This is a somewhat
limiting but extremely useful property known as [parametricity](https://en.wikipedia.org/wiki/Parametricity).
This implies, that a function should behave in same way for all values,
i.e. inspection of value is not allowed.** for instance, that code that
behaves differently for the `null` or `undefined` value than for other
values ~~is unlikely to comply~~ complying with the rule.


we can steal some ideas from https://wiki.haskell.org/Polymorphism or/and link to it too (has better explanation then wikipedia)

Reply:
I would like to expand this section.  These are good ideas.  The "unless otherwise specified" has to do with things like `f must be a binary function` or `b must be same Apply as a`.  In such cases, not every value is allowable.

"is unlikely to comply" was about examples such as  `return r === null ? Maybe.of(r) : Maybe.of(r)`.  The only way these examples could work is if their differences are not observable, so perhaps we can skip it.  But it was part of the motivation for this change.